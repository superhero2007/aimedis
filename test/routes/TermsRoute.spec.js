import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import TermsRoute from 'routes/TermsRoute';

describe('routes/TermsRoute', () => {
  it('should render the Terms Page for unauthenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/']}>
        <TermsRoute
          exact
          path="/"
          component={() => (<div>Terms</div>)}
          isAuthenticated={false}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });

  it('should render the Terms Page for authenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/']}>
        <TermsRoute
          exact
          path="/"
          component={() => (<div>Terms</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
