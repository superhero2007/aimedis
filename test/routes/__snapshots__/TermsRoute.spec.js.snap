// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/TermsRoute should render the Terms Page for authenticated access 1`] = `"<div data-reactroot=\\"\\">Terms</div>"`;

exports[`routes/TermsRoute should render the Terms Page for unauthenticated access 1`] = `"<div data-reactroot=\\"\\">Terms</div>"`;
